src/silo_api/command_line_arguments.cpp

```diff
@@ -1,26 +1,66 @@
 #include "silo_api/command_line_arguments.h"
 
+#include <boost/algorithm/string/join.hpp>
+
 using silo_api::CommandLineArguments;
 
+std::string CommandLineArguments::asUnixOptionString(
+   const silo::config::AbstractConfigSource::Option& option
+) {
+   std::vector<std::string> result;
+   for (const std::string& current_string : option.access_path) {
+      std::string current_result;
+      for (const char character : current_string) {
+         if (std::isupper(character)) {
+            current_result += '-';
+            const char char_in_lower_case =
+               static_cast<char>(std::tolower(static_cast<unsigned char>(character)));
+            current_result += char_in_lower_case;
+         } else {
+            current_result += character;
+         }
+      }
+      result.emplace_back(current_result);
+   }
+   return "SILO_" + boost::join(result, "-");
```

Comment:
This is the result of `siloApi --help`:

```
usage: siloApi OPTIONS
SILO - Sequence Indexing engine for Large Order of genomic data

--daemon                                                     Run application 
                                                             as a daemon.
--umask=mask                                                 Set the daemon's 
                                                             umask (octal, 
                                                             e.g. 027).
--pidfile=path                                               Write the process
                                                             ID of the 
                                                             application to 
                                                             given file.
-h, --help                                                   display help 
                                                             information on 
                                                             command line 
                                                             arguments
--preprocessingConfig=PATH                                   path to the 
                                                             preprocessing 
                                                             config file
--databaseConfig=PATH                                        path to the 
                                                             database config 
                                                             file
-dPATH, --SILO_data-directory=PATH                           path to the 
                                                             preprocessed data
--SILO_port=NUMBER                                           port to listen to
                                                             requests
--maxQueuedHttpConnections=NUMBER                            maximum number of
                                                             http connections
--threadsForHttpConnections=NUMBER                           number of threads
                                                             for http 
                                                             connections
-a, --api                                                    Execution mode: 
                                                             start the SILO 
                                                             web interface
-p, --preprocessing                                          Execution mode: 
                                                             trigger the 
                                                             preprocessing 
                                                             pipeline to 
                                                             generate a 
                                                             partitioned 
                                                             dataset that can 
                                                             be read by the 
                                                             database
-tMINUTES, --SILO_estimated-startup-time-in-minutes=MINUTES  Estimated time in
                                                             minutes that the 
                                                             initial loading 
                                                             of the database 
                                                             takes. As long as
                                                             no database is 
                                                             loaded yet, SILO 
                                                             will throw a 503 
                                                             error. This 
                                                             option allows 
                                                             SILO to compute a
                                                             Retry-After 
                                                             header for the 
                                                             503 response. 
``` 

* IMO the `SILO_` prefix looks weird. Should we get rid of it?
* The casing is not consistent (`SILO_estimated-startup-time-in-minutes` vs `maxQueuedHttpConnections`. Depending on which casing style we go for, this will be another breaking change. Actually I prefer to leave it camel case, or are there good arguments to make everything kebap case?

Reply:
The SILO prefix is mistakenly added. I am still thinking of a way to have these options nicely displayed automatically. I think that we want to

Reply:
(the SILO prefix is only meant for environment variables)

Reply:
Christian mentioned that kebab-case is the defacto UNIX standard